com.vrcfury.vrcfury/Editor/VF/Feature/CombineMeshesBuilder.cs

```diff
@@ -0,0 +1,543 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Animations;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.UIElements;
+using VF.Builder;
+using VF.Feature.Base;
+using VF.Inspector;
+using VF.Model.Feature;
+using VF.Utils;
+using VF.Injector;
+using VF.Service;
+using VRC.SDKBase;
+using static System.Linq.Enumerable;
+
+namespace VF.Feature
+{
+    public class CombineMeshesBuilder : FeatureBuilder<CombineMeshes>
+    {
+        [VFAutowired] private readonly ObjectMoveService mover;
+
+        private static string[] preferredBaseMeshNames = new string[] { "body", "main" };
+
+        public override string GetEditorTitle()
+        {
+            return "Combine Meshes";
+        }
+
+        public override VisualElement CreateEditor(SerializedProperty prop)
+        {
+            var content = new VisualElement();
+            content.Add(VRCFuryEditorUtils.Info(
+                "This feature will combine all similar meshes that are not animated to be toggled into one, " +
+                " saving extra draw calls!"
+            ));
+            content.Add(new VisualElement { style = { paddingTop = 10 } });
+            content.Add(VRCFuryEditorUtils.Prop(prop.FindPropertyRelative("keepFaceMeshSeparate"), "Keep Face Mesh separate"));
+            content.Add(new VisualElement { style = { paddingTop = 10 } });
+            return content;
+        }
+
+        public override bool AvailableOnProps()
+        {
+            return false;
+        }
+
+        [FeatureBuilderAction(FeatureOrder.BlendshapeOptimizer)]
+        public void Apply()
+        {
+            var changes = CalculateMeshChanges();
+            foreach (var group in changes.Groups)
+            {
+                var skins = group.Value;
+                var basis = skins[0];
+                VFGameObject basisObject = basis.gameObject;
+                var meshesToAdd = skins.Skip(1).ToList();
+                basis.sharedMesh = mutableManager.MakeMutable(basis.sharedMesh, basis.owner()); ;
+                VRCFuryEditorUtils.MarkDirty(basis);
+
+                var combinable = new CombinableMesh(basis);
+                foreach (var mesh in meshesToAdd)
+                {
+                    combinable.AddMesh(mesh);
+                    VFGameObject obj = mesh.gameObject;
+                    mover.DirectRewrite(obj, basisObject);
```

Comment:
Is this safe? If there's any animations that would be rewritten here, wouldn't those things prevent the meshes from being joined to begin with?